homeassistant/components/arcticspa/__init__.py

```diff
@@ -0,0 +1,38 @@
+"""The Arctic Spa integration."""
+from __future__ import annotations
+
+from http import HTTPStatus
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import ArcticSpaDataUpdateCoordinator
+from .hottub import Device
+
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up an Arctic Spa from a config entry."""
+
+    device = Device(entry.data[CONF_API_KEY])
+    code = await device.async_authenticate()
+    if code != HTTPStatus.OK:
+        return False
```

Comment:
Why does this happen? Should we differentiate?

You can raise ConfigEntryNotReady when HA should try again later, or ConfigEntryAuthFailed when the API key expired (but this requires reauth flow so this one is out of the picture for now to keep the PR small)

Are you the owner of the lib? If so maybe it's an idea to raise different errors on these events to catch them properly so you don't have to compare http status codes here

Reply:
Thanks for the pointers. I wrote the underlying library. I implemented better error-handling there and updated this code too.